### ISSUE TITLE
Defining constant with __FILE__ in name breaks execution

### ISSUE THREAD

**Issue by aaemnnosttv** [wp-cli/wp-cli#5332]
## Bug Report

- [x] Yes, I reviewed the [contribution guidelines](https://make.wordpress.org/cli/handbook/contributing/).
- [x] Yes, more specifically, I reviewed the guidelines on [how to write clear bug reports](https://make.wordpress.org/cli/handbook/bug-reports/).

**Describe the current, buggy behavior**

Defining a constant in `wp-config.php` that includes `__FILE__` in the _name_ of the constant, causes wp-cli to fail with this error message:

```
Parse error: syntax error, unexpected 'var' (T_VAR) in phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1197) : eval()'d code on line 66
```

Using latest v2.4.0 phar executable.

**Describe how other contributors can replicate this bug**

1. Add this line to `wp-config.php`  
    ```php
    define( 'WP_CONFIG__FILE__', __FILE__ );
    ```
1. Run `wp`
1. See the above error, command fails

**Describe what you expect as the correct outcome**

I would expect the wp command list to display.

**Let us know what environment you are running this on**

```
OS:	Linux 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 x86_64
Shell:	
PHP binary:	/usr/local/bin/php
PHP version:	7.3.9
php.ini used:	/usr/local/etc/php/php.ini
WP-CLI root dir:	phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/var/www/html
WP-CLI packages dir:	
WP-CLI global config:	
WP-CLI project config:	
WP-CLI version:	2.4.0
```

**Comment by schlessera** [comment]
The code in https://github.com/wp-cli/wp-cli/blob/d68399bc8d5a909c2c7f9f6c4483c9c7b8d0f5c3/php/utils.php#L614-L624 is too primitive and easily breaks.

We should use the code from https://github.com/wp-cli/eval-command/pull/44 and provide it as an improved centralized helper here, and then make `eval-file` reuse that helper.

- [ ] Port code from `eval-file` to `wp-cli/wp-cli`
- [ ] *After next `wp-cli/wp-cli` release:* Use `Utils/replace_path_consts()` in `eval-file`
- [ ] *After next `wp-cli/wp-cli` release:* Bump version requirement of `wp-cli/wp-cli` in `wp-cli/eval-command`

**Comment by schlessera** [comment]
This should be resolved with latest `master` and fixed in upcoming v2.5.0 release.
